query/README.md

```diff
@@ -21,35 +21,183 @@ KeenQueryClient queryClient = new TestKeenQueryClientBuilder(queryProject)
+        .build();
+QueryResult result = queryClient.execute(query, new Timeframe("this_month"));
+if (result.isInteger()) {
+	Integer countValue = result.getInteger();
+	// do something with countValue
+}
+```
+
+Some special cases are when filters "Group By" and "Inteval" are specified, as well as the Select Unique query.
+
+For Select Unique queries, the user gets a list of unique values, given the target property. Therefore, the QueryResult will be a list of unique property values. The QueryResult type only supports Integer, Double, String, and ArrayList values; therefore, if the property value is not of the aforementioned types, then it may be of generic Object type.
+
+``` java
+Query query = new QueryBuilder(QueryType.SELECT_UNIQUE_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withTargetProperty("click-number")
+        .build();
+QueryResult result = queryClient.execute(query, new Timeframe("this_month"));
+if (result.isList()) {
+	ArrayList<QueryResult> listResults = result.getList();
+	for (QueryResult item : listResults) {
+		if (item.isInteger()) {
+			// do something with Integer value
+		}
+		
+		// note that for Select Unique query, QueryResult can also be a generic Object,
+		// depending on the expected type of the Target Property.
+		if (item.isObject()) {
+			// in this case, user is responsible for her own parsing.
+		}
+	}
+}
+```
+
+Filtering any query via Group-By will cause the query response to consist of an ArrayList of GroupBy objects. Each GroupBy object contains a HashMap<String, Object> of property/values, as well as a QueryResult of the result.
+``` java
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withGroupBy("click-number")
+        .build();
+QueryResult result = queryClient.execute(query, new Timeframe("this_month"));
+if (result.isList()) {
+	ArrayList<QueryResult> listResults = result.getList();
+	for (QueryResult item : listResults) {
+		if (item.isGroupBy()) {
+			GroupBy groupBy = item.getGroupBy();
+			HashMap<String, Object> properties = groupBy.getProperties();
+			QueryResult groupByValue = groupBy.getResult();
+			if (groupByValue.isInteger()) {
+				// do something with integer result.
+			}
+		}
+	}
+}
+```
+Filtering any query via Interval will cause the query response to consist of an ArrayList of Interval objects. Each Interval object contains a Timeframe for the interval, as well as a QueryResult of the value.
+``` java
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
         .withEventCollection("<event_collection>")
-        .withRelativeTimeframe("this_month")
+        .withInterval("weekly")
+        .build();
+QueryResult result = queryClient.execute(query, new Timeframe("this_year"));
+if (result.isList()) {
+	ArrayList<QueryResult> listResults = result.getList();
+	for (QueryResult item : listResults) {
+		if (item.isInterval()) {
+			Interval interval = item.getInterval();
```

Comment:
Ahhh, great idea! I'll have GroupByResult and IntervalResult extend QueryResult.